treeherder/webapp/api/performance_serializers.py

```diff
@@ -116,8 +117,65 @@ class Meta:
         ]
 
 
+class AlertTaskclusterMetadataSerializer(serializers.ModelSerializer):
+    current_push = serializers.SerializerMethodField()
+    prev_push = serializers.SerializerMethodField()
+
+
+    def get_current_push(self, alert):
+        try:
+            datum_q = PerformanceDatum.objects.filter(
+                signature=alert.series_signature,
+                repository=alert.series_signature.repository,
+                push_id=alert.summary.push
+            )
+            if len(datum_q):
+                datum = datum_q.first()
+                metadata = TaskclusterMetadata.objects.get(job=datum.job)
+                return {
+                    'task_id': metadata.task_id,
+                    'retry_id': metadata.retry_id,
+                    'result': datum.job.result,
+                }
+            else:
+                return {}
+        except ObjectDoesNotExist:
+            return {}
+
+    def get_prev_push(self, alert):
+        try:
+            datum_q = PerformanceDatum.objects.filter(
+                signature=alert.series_signature,
+                repository=alert.series_signature.repository,
+                push_id=alert.summary.prev_push
+            )
+            if len(datum_q):
+                datum = datum_q.first()
+                metadata = TaskclusterMetadata.objects.get(job=datum.job)
+                return {
+                    'task_id': metadata.task_id,
+                    'retry_id': metadata.retry_id,
+                    'result': datum.job.result,
+                }
+            else:
+                return {}
+        except ObjectDoesNotExist:
```

Comment:
Doesn't `filter` return an empty queryset if no results are found? The try block might not be needed here. I am only aware of `get()` throwing an [DoesNotExist](https://docs.djangoproject.com/en/4.2/ref/models/class/#django.db.models.Model.DoesNotExist) exception.